Source/JavaScriptCore/wasm/WasmBBQJIT64.cpp

```diff
@@ -2310,8 +2310,12 @@ PartialResult WARN_UNUSED_RETURN BBQJIT::addI64Add(Value lhs, Value rhs, Value&
             m_jit.add64(lhsLocation.asGPR(), rhsLocation.asGPR(), resultLocation.asGPR());
         ),
         BLOCK(
-            m_jit.move(ImmHelpers::regLocation(lhsLocation, rhsLocation).asGPR(), resultLocation.asGPR());
-            m_jit.add64(TrustedImm64(ImmHelpers::imm(lhs, rhs).asI64()), resultLocation.asGPR());
+            if (LIKELY(isRepresentableAs<int32_t>(ImmHelpers::imm(lhs, rhs).asI64())))
+                m_jit.add64(TrustedImm32(ImmHelpers::imm(lhs, rhs).asI64()), ImmHelpers::regLocation(lhsLocation, rhsLocation).asGPR(), resultLocation.asGPR());
+            else {
+                m_jit.move(ImmHelpers::regLocation(lhsLocation, rhsLocation).asGPR(), resultLocation.asGPR());
```

Comment:
I don't understand why this move becomes necessary when the value is >32b wide, what's the reasoning there?

Reply:
Since so far we only have
```
void add64(TrustedImm32 imm, RegisterID src, RegisterID dest)
```
no
```
void add64(TrustedImm64 imm, RegisterID src, RegisterID dest)
```

Reply:
Since so far we only have 
```
void add64(TrustedImm32 imm, RegisterID src, RegisterID dest)
```
no
```
void add64(TrustedImm64 imm, RegisterID src, RegisterID dest)
```
for x86_64.